### DISCUSSION TITLE
How to using globbing backticks for expressions?

### DISCUSSION THREAD

**Discussion by halloleo** [xonsh/xonsh#4525] [Q&A] [answered]
How can I use the globbing backticks 

    `...`

and

    g`...`

for expressions?

How can I construct a globbing expression from a (Python) string in xonsh? Say, I have a directory in the Python var `d` and want to find the `*.py` files in `d`. If I want to use the backtick syntax I would need something like

g`@(d+'/*.py')`

but that obviously does not work...

How can I do it?

**Comment by agoose77** [comment, marked as answer]
Interestingly, we don't seem to support f-globstrings at the moment. I feel that we _probably_ should support this, given that Python-like f-strings work in subproc mode:
```bash
x = 1 # Python
echo f"{x}" # Subproc
```

In the meantime, we could use a [custom glob command](https://github.com/anki-code/xonsh-cheatsheet/blob/main/README.md#globbing---get-the-list-of-files-from-path-by-mask-or-regexp) do do this, or manually create the glob string and pass it to `__xonsh__.globsearch` or `__xonsh__.glob`.

**Reply by halloleo** [reply]
Thanks @agoose77 ! Do Iunderstand you correctly that at this stage I can write:

    ls @(__xonsh__.globpath(d+'/*.py'))

and get all python files in the directory `d`?

**Reply by agoose77** [reply]
Nearly; I made a typo in the original reply. You want `__xonsh__.glob` or `__xonsh__.globsearch`.

**Reply by halloleo** [reply]
Thanks for the correction. Works like a treat!

**Reply by agoose77** [reply]
Note that you can also use python mode here, something like

`echo @(Path(d).glob(...))`
